cmd/zfs/zfs_main.c

```diff
@@ -6823,6 +6840,104 @@ zfs_do_bookmark(int argc, char **argv)
 	return (-1);
 }
 
+static int
+zfs_do_key(int argc, char **argv)
+{
+	int c, ret = -1;
+	boolean_t load = B_FALSE, unload = B_FALSE, rewrap = B_FALSE;
+	nvlist_t *props = NULL;
+	zfs_handle_t *zhp = NULL;
+
+	if (nvlist_alloc(&props, NV_UNIQUE_NAME, 0) != 0)
```

Comment:
I was never completely sure when looking at other code. When would it be appropriate to use the `fnvlist_*()` functions vs the regular `nvlist_*()` variants?